src/google/protobuf/compiler/objectivec/objectivec_helpers.cc

```diff
@@ -243,21 +243,36 @@ bool IsReservedCIdentifier(const string& input) {
   return false;
 }
 
-string SanitizeNameForObjC(const string& input,
+string SanitizeNameForObjC(const string& prefix,
+                           const string& input,
                            const string& extension,
                            string* out_suffix_added) {
   static const std::unordered_set<string> kReservedWords =
       MakeWordsMap(kReservedWordList, GOOGLE_ARRAYSIZE(kReservedWordList));
   static const std::unordered_set<string> kNSObjectMethods =
       MakeWordsMap(kNSObjectMethodsList, GOOGLE_ARRAYSIZE(kNSObjectMethodsList));
-  if (IsReservedCIdentifier(input) ||
-      (kReservedWords.count(input) > 0) ||
-      (kNSObjectMethods.count(input) > 0)) {
+  string sanitized;
+  // If the string already has a prefix, don't add it again.
```

Comment:
Can you fix up these comments since we do add it in some cases?

You can also collapse the first two into:

```
if (input.length() == prefix.length() || !ascii_isupper(input[prefix.length()])) {
  sanitized = prefix + input;
} else {
  sanitized = input;
}
```

unless documenting it is easier to keep them split.